### PR TITLE
Use global state context instead of drilling down props.

### PR DESCRIPTION
**Pull request by lublagg**
Before working on adding a hook for listening to and updating CODAP state, I thought it would be useful to also update the way we manage state so that we don't have to drill down props as much and can just create functions where they are needed.

### CODE REVIEW

**Review by eireland — CHANGES_REQUESTED** [review]
Nice job on the refactoring! Found a few more places that may be could be refactored. I was going back and forth on whether to move the spinner specific handlers into the spinner view (`handleStartDrag`, and `handleDeleteWege`), but I'll leave it up to you.

**Comment by eireland** [comment]
Yes! Thank you for doing this. I was thinking the same thing when we started work on it.

**Comment by dougmartin** [comment]
Skipping this review since @eireland already reviewed it.

File: src/components/model/column.tsx [hunk 1]
```diff
@@ -1,103 +1,41 @@
...
-        attrNameInputRef.current?.blur();
-        handleNameChange(deviceId, e.currentTarget.value);
-        break;
-    }
-  };
+export const Column = ({column, columnIndex}: IProps) => {
+  const { globalState } = useGlobalStateContext();
+  const { model } = globalState;
+  const hasBranch = model.columns.find( c => c.id === column.id && c.devices.length > 1);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (eireland) [inline src/components/model/column.tsx:20]: Remove unused var disable since it is no longer used.

**Review by eireland — APPROVED** [review]
Looks good. I did leave a comment re trimming column names so that checking for uniqueness is easier.

File: src/components/model/column-header.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  const { model } = globalState;
+  const [columnName, setColumnName] = useState(column.name);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setColumnName(column.name);
+  }, [column.name]);
+
+  const handleNameChange = async () => {
+    const newName = getNewColumnName(columnName, model.columns, column.id);
```
[the remark this answers is not in the document]
Comment (lublagg) [reply src/components/model/column-header.tsx]: Done, good catch :)

**Pull request merged**